services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py

```diff
@@ -178,6 +179,19 @@ async def get_job_outputs(
     node_ids = list(project.workbench.keys())
     assert len(node_ids) == 1  # nosec
 
+    product_price = await webserver_api.get_product_price()
+    if product_price is not None:
+        wallet = await webserver_api.get_project_wallet(project_id=project.uuid)
+        if wallet is None:
+            raise MissingWallet(
+                f"Job {project.uuid} does not have an associated wallet."
+            )
+        wallet_with_credits = await webserver_api.get_wallet(wallet_id=wallet.wallet_id)
+        if wallet_with_credits.available_credits < 0.0:
```

Comment:
Q: why do we release this information when credits are equal to 0? Should they not be positive?

Reply:
Here, the simulation has already been run. So this is just checking the simulation didn't use more credits than were available. I.e. I am just checking the user is not in debt 😄